nautobot/core/ui/object_detail.py

```diff
@@ -814,9 +820,25 @@ def get_extra_context(self, context: Context):
             if self.order_by_fields:
                 body_content_table_queryset = body_content_table_queryset.order_by(*self.order_by_fields)
             body_content_table_queryset = body_content_table_queryset.distinct()
-            body_content_table = body_content_table_class(
-                body_content_table_queryset, hide_hierarchy_ui=self.hide_hierarchy_ui
-            )
+            if self.tab_id:
+                # Use the `self.tab_id`, if it exists, to determine the correct return URL for the table
+                # to redirect the user back to the correct tab after editing/deleteing an object
+                body_content_table = body_content_table_class(
+                    body_content_table_queryset,
+                    hide_hierarchy_ui=self.hide_hierarchy_ui,
+                    extra_columns=[
+                        (
+                            "actions",
+                            ButtonsColumn(
+                                model=body_content_table_queryset.model, return_url_extra=f"?tab={self.tab_id}"
+                            ),
+                        )
+                    ],
```

Comment:
This feels wrong to me... can we instead allocate the table as before and just modify `body_content_table.columns["actions"]` directly instead?

Reply:
Apparently, we cannot modify `body_content_table.columns`, but we can with `body_content_table.base_columns`. That achieved the same effect.